api/tools/src/main/java/org/pmiops/workbench/tools/LoadDataDictionary.java

```diff
@@ -41,41 +43,48 @@ public CommandLineRunner run(
 
       for (Resource resource : getDataDictionaryExportFiles()) {
         InputStream is = resource.getInputStream();
+        ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+        SimpleDateFormat df = new SimpleDateFormat("dd-MM-yyyy hh:mm:ss");
+        mapper.setDateFormat(df);
 
-        BufferedReader reader = new BufferedReader(new InputStreamReader(is));
-        Timestamp definedTime = new Timestamp(Long.parseLong(reader.readLine()) * 1000);
+        mapper.registerModule(new KotlinModule());
+        DataDictionary dd = mapper.readValue(is, DataDictionary.class);
 
-        String line;
-        while ((line = reader.readLine()) != null) {
-          String[] fields = line.split(":");
+        Timestamp newEntryDefinedTime = dd.getMeta_data()[0].getCreated_time();
 
-          String relevantOmopTable = fields[0];
-          String fieldName = fields[1];
+        CdrVersion cdrVersion = cdrVersionDao.findByName(dd.getMeta_data()[0].getCdr_version());
 
+        if (cdrVersion == null) {
+          // Skip over Data Dictionaries for CDR Versions not in the current environment
+          continue;
+        }
+
+        for (AvailableField field : dd.getTransformations()[0].getAvailable_fields()) {
           DataDictionaryEntry entry =
               dataDictionaryEntryDao.findByRelevantOmopTableAndFieldNameAndCdrVersion(
-                  relevantOmopTable, fieldName, defaultCdrVersion);
+                  field.getRelevant_omop_table(), field.getField_name(), cdrVersion);
 
           // We are skipping ahead if the defined times match by assuming that the definition has
           // not changed.
-          if (entry != null && definedTime.before(entry.getDefinedTime())) {
+          if (entry != null && newEntryDefinedTime.before(entry.getDefinedTime())) {
             continue;
           }
 
           if (entry == null) {
             entry = new DataDictionaryEntry();
-            entry.setRelevantOmopTable(relevantOmopTable);
-            entry.setFieldName(fieldName);
-            entry.setCdrVersion((defaultCdrVersion));
+            entry.setRelevantOmopTable(field.getRelevant_omop_table());
```

Comment:
nit: is there a way to chain these or define a builder for DDE?

Reply:
I'm inclined to say it's not worth the effort for 1 time use.

Reply:
I thought we got that for free with Kotlin. But from what I'm reading, only Kotlin clients can take advantage of named parameters.

Reply:
Anyway, I'd argue builders are worth it for single-time use, especially if you can enforce required parameters. Not necessary at this point in the game for this PR though.